aws-storage-s3/src/main/java/com/amplifyframework/storage/s3/transfer/TransferWorkerObserver.kt

```diff
@@ -186,8 +186,15 @@ internal class TransferWorkerObserver private constructor(
                 )
             )?.use {
                 while (it.moveToNext()) {
-                    val id = it.getInt(it.getColumnIndexOrThrow(TransferTable.COLUMN_ID))
-                    attachObserver(id.toString())
+                    // getInt has been reported to  throw in rare cases where it appears another thread
+                    // may have closed cursor
+                    try {
+                        val id = it.getInt(it.getColumnIndexOrThrow(TransferTable.COLUMN_ID))
+                        attachObserver(id.toString())
+                    } catch (exception: IllegalStateException) {
+                        // Stop iterating over cursor. Cursor is likely closed
+                        break
```

Comment:
Is this situation worthy of an error long? What is the impact of failing to attach an observer?